internal/ask/form/aggregation.go

```diff
@@ -0,0 +1,429 @@
+package form
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+
+	"github.com/ardanlabs/kit/log"
+	"github.com/coralproject/shelf/internal/ask/form/submission"
+	"github.com/coralproject/shelf/internal/platform/db"
+	"gopkg.in/mgo.v2/bson"
+)
+
+//==============================================================================
+
+// MCAnswerAggregation holds the count for selections of a single multiple
+// choice answer.
+type MCAnswerAggregation struct {
+	Title string `json:"answer" bson:"answer"`
+	Count int    `json:"count" bson:"count"`
+}
+
+// MCAggregation holds a multiple choice question and a map aggregated counts for
+// each answer. The Answers map is keyed off an md5 of the answer as not better keys exist
+type MCAggregation struct {
+	Question  string                         `json:"question" bson:"question"`
+	MCAnswers map[string]MCAnswerAggregation `json:"answers" bson:"answers"`
+}
+
+// TextAggregation holds the aggregated text based answers for a single question
+// marked with the Incude in Aggregations tag, orderd by [question_id][answer].
+type TextAggregation map[string]string
```

Comment:
Naming this type provides a lot of clarity in usage below. It makes it explicit what this map is for.